dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/MultiUserReferenceParameter.java

```diff
@@ -0,0 +1,113 @@
+        super(key, displayName, defaultValue, isRequired);
+    }
+
+    @Override
+    public String hasError(final String stringToValidate) {
+        final StringBuffer errorMsg = new StringBuffer();
+        if (UtilMethods.isSet(stringToValidate)) {
+            final StringTokenizer tokenizer = new StringTokenizer(stringToValidate, ",");
+            while (tokenizer.hasMoreTokens()) {
+                final String token = tokenizer.nextToken().trim();
+                if (Validator.isEmailAddress(token)) {
+                    try {
+                        APILocator.getUserAPI()
+                                .loadByUserByEmail(token, APILocator.getUserAPI().getSystemUser(),
+                                        false);
+                    } catch (Exception e) {
+                        Logger.error(this.getClass(), "Unable to find user with email: " + token);
+                        errorMsg.append("Unable to find user with email: " + token + "</br>");
+                    }
+                } else {
+                    String error = null;
+                    if (isUserId(token)) {
+                        continue;
+                    } else {
+                        error = "Unable to find user with ID: " + token + "</br>";
+                        if (isRoleKey(token)) {
+                            continue;
+                        } else {
+                            error = (UtilMethods.isSet(error)) ? error
+                                    : "Unable to find users assigned to role key: " + token;
+                        }
+                        errorMsg.append(error);
+                    }
+                }
+            }
+        }
+        return (UtilMethods.isSet(errorMsg.toString())) ? errorMsg.toString() : null;
+    }
+
+    /**
+     * Checks whether the specified value belongs to an existing user ID or not.
+     *
+     * @param userId The potential user ID.
+     *
+     * @return Returns {@code true} if the specified value is a valid user ID. Otherwise, returns
+     * {@code false}.
+     */
+    private boolean isUserId(final String userId) {
+        try {
+            APILocator.getUserAPI()
+                    .loadUserById(userId, APILocator.getUserAPI().getSystemUser(), false);
+            return Boolean.TRUE;
+        } catch (Exception e) {
+            return Boolean.FALSE;
+        }
+    }
+
+    /**
+     * Checks whether the specified value belongs to an existing role key or not.
+     *
+     * @param roleKey The potential role key.
+     *
+     * @return Returns {@code true} if the specified value is a valid role key. Otherwise, returns
+     * {@code false}.
+     */
+    private boolean isRoleKey(final String roleKey) {
+        try {
+            final Role role = APILocator.getRoleAPI().loadRoleByKey(roleKey);
+            APILocator.getRoleAPI()
+                    .findUsersForRole(role);
+            return Boolean.TRUE;
+        } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)